enhancements/installer/component-selection.md

```diff
@@ -0,0 +1,500 @@
+small set of profiles to define specific sets of components to be included, not to allow a user to fully pick
+and choose individual components.  We would have to define a large set of profiles to encompass all the possible
+combinations of components to be enabled/disabled.  Direction has been set that cluster profiles should only be
+used defining the standalone and hypershift form factors.
+
+* CVO already supports an EXCLUDE_MANIFESTS env variable which is used to implement the ROKS deployment topology.
+Unfortunately it only allows a single annotation to be specified, so even if we want to use it for this purpose
+it needs to be extended to support multiple annotations so multiple individual components can be excluded
+independently rather than requiring all components to be excluded to share a single common annotation.
+
+Regardless we need a way to expose this configuration as a first class part of the install config provided by the
+user creating the cluster, so at a minimum we need to add a mechanism to wire an install config value into
+the CVO arguments and allow the CVO to consume more than a single annotation to exclude.
+
+* Allow the installer to specify additional resources to `include` in addition to ones to `exclude`.  This has the challenge
+of potentially conflicting with the specific set of resources that a cluster_profile defines.  There are some
+components that should never be deployed in a particular cluster_profile and so we do not want to allow the user
+to add them.  Examples would be resources that should only be created in standalone installs, not hypershift
+managed ones, because hypershift has its own versions of those resources.
+
+* Use clusteroverrides to exclude content.  The problem w/ this approach is it puts the cluster into an unsupported
+and non-upgradeable state.
+
+* Move "optional" CVO components to be OLM based and make it possible to install+upgrade OLM operators as though
+they are part of the payload.  While this is part of our longterm roadmap, it has two major challenges.  First,
+moving a component from the CVO to OLM is likely more engineering work and has more significant implications on
+the distribution+testing of that component, than adding this filtering capability to the CVO, so it is an easier
+win in the short term.  Secondly, OLM itself needs significant new functionality to be able to install+upgrade
+OLM operators as part of the cluster lifecycle.  The long timeline it will take to build that capability makes
+it necessary that we take a tactical approach in the short term to enable this capability at the CVO level.
+
+* If we want components to be disabled by default (either existing components or new ones added in the future),
+we can add their component names to a list of default-disabled components that the installer populates
+the install-config with.  Users can then edit that disabled list in the install-config to remove those
```

Comment:
I think the current body of the enhancement has the installer performing opaque pass-through without doing the sort of release-inspection or whatever would be needed to give it a list of available components.  Floated somewhere in past discussion was freezing out hard-coded lists of capabilities so the installer could complain about invalid values (hooray typo catching, but not sure it's worth the maintenance cost).  Have we made a decision about the details of the installer's planned behavior?  If so, can we doc them in the body?  And then update this alternative entry to explain why we chose that route instead of the other possible installer approaches?

Reply:
the current EP says we'll bake the list of valid capabilities into the release image (via logic in `oc adm release new`), it takes no final position on either:

1) the installer validating the include/exclude values in the install-config, against those values in the release-image (I could see us deciding to do this but we can have an MVP w/o it)
2) the installer pre-populating any include/exclude values in the generated install-config under any circumstances (this seems less likely, certainly not in the first pass)

Reply:
[this thread][1] has us settling on the installer validating values in the install-config.  Although I guess we don't have to officially commit to that if we wanted to leave it as an underpromise/overdeliver or follow-up pass.

[1]: https://github.com/openshift/enhancements/pull/922#discussion_r779146272

Reply:
The existence of the "Default include/exclude" functionality does remove the concern that validation would prevent an admin from pre-emptively excluding a future capability, so I guess we might as well have the installer validate.  I will mention this.